### PR TITLE
feat(gcs): support import grants for gcs provider

### PR DESCRIPTION
**Pull request by rahmatrhd**

### CODE REVIEW

**Comment by mabdh** [comment]
@rahmatrhd looks like there is a flaky test there

File: plugins/providers/gcs/client.go [hunk 1]
```diff
@@ -84,3 +80,48 @@ func (c *gcsClient) RevokeBucketAccess(ctx context.Context, b Bucket, identity s
...
+				m := strings.Split(member, ":")
+				if len(m) == 0 || len(m) > 2 {
+					return nil, fmt.Errorf("invalid bucket access member signature %q", member)
+				} else if len(m) == 2 {
+					accountID = m[1]
+				}
+				accountType = m[0]
+
+				// exclude unsupported account types
+				excludedAccountTypes := []string{"allUsers", "allAuthenticatedUsers", "projectOwner", "projectEditor", "projectViewer"}
```
Comment (haveiss) [inline plugins/providers/gcs/client.go:108]: since this is static, how about moving this to a constant

Comment (mabdh) [inline plugins/providers/gcs/client.go:108]: can this be moved as var in package level?

File: plugins/providers/gcs/client.go [hunk 2]
```diff
@@ -84,3 +80,48 @@ func (c *gcsClient) RevokeBucketAccess(ctx context.Context, b Bucket, identity s
...
+		for _, role := range policy.Roles() {
+			for _, member := range policy.Members(role) {
+				var accountType, accountID string
+				m := strings.Split(member, ":")
+				if len(m) == 0 || len(m) > 2 {
+					return nil, fmt.Errorf("invalid bucket access member signature %q", member)
+				} else if len(m) == 2 {
+					accountID = m[1]
+				}
+				accountType = m[0]
```
Comment (haveiss) [inline plugins/providers/gcs/client.go:105]: let's move this to separate function to make it easier to read and less responsibility on this function

File: plugins/providers/gcs/provider_test.go [hunk 3]
```diff
@@ -828,6 +829,53 @@ func TestGetAccountType(t *testing.T) {
...
+func TestListAccess(t *testing.T) {
+	crypto := new(mocks.Crypto)
+	client := new(mocks.GCSClient)
+	p := gcs.NewProvider("gcs", crypto)
+	providerURN := "test-resource-name"
+	p.Clients = map[string]gcs.GCSClient{
+		providerURN: client,
+	}
+
+	pc := &domain.ProviderConfig{
```
Comment (haveiss) [inline plugins/providers/gcs/provider_test.go:841]: let's use clear and meaningful variable names to improve readability

File: plugins/providers/gcs/provider_test.go [hunk 4]
```diff
@@ -828,6 +829,53 @@ func TestGetAccountType(t *testing.T) {
...
+						Description: "Read access to buckets with object listing/creation/deletion",
+						Permissions: []interface{}{"roles/storage.legacyBucketWriter"},
+					},
+				},
+			},
+		},
+	}
+
+	crypto.EXPECT().
+		Decrypt("c2VydmljZV9hY2NvdW50LWtleS1qc29u").Return("service_account-key-json", nil).Once()
```
Comment (haveiss) [inline plugins/providers/gcs/provider_test.go:864]: let's avoid hard-coded values

```go
encryptedSAKey := base64.StdEncoding.EncodeToString([]byte("service_account-key-json"))
providerConfig := &domain.ProviderConfig{
                ...
		Credentials: gcs.Credentials{
			ServiceAccountKey: encryptedSAKey,
			...
		},
		...
	}

	crypto.EXPECT().
		Decrypt(encryptedSAKey).Return("service_account-key-json", nil).Once()
```

**Review by haveiss — APPROVED** [review]

**Pull request merged**